src/config.rs

```diff
@@ -558,7 +558,9 @@ impl Config {
             self.get_path().join("DO_NOT_USE_THIS_DIRECTORY_FOR_ANYTHING"),
         );
 
-        self.try_lock(options.open(&self.db_path())?)
+        let file = self.try_lock(options.open(&self.db_path())?)?;
+        fs::File::open(self.get_path())?.sync_all()?;
+        Ok(file)
```

Comment:
Ahh interesting, this makes a lot of sense in hindsight, but I don't think I've ever thought to fsync a file that I've just opened before!

Reply:
This fsync is actually on the directory containing the file, so it doesn't disappear. I'll move this into a function call, so it will be more clear.